lib/Sema/TypeCheckType.cpp

```diff
@@ -3159,9 +3158,7 @@ Type TypeResolver::resolveArrayType(ArrayTypeRepr *repr,
 
   auto sliceTy =
     TypeChecker::getArraySliceType(repr->getBrackets().Start, baseTy);
-  if (!sliceTy)
-    return ErrorType::get(Context);
-
+  assert(sliceTy && "TypeChecker::getArraySliceType returned nullptr.");
```

Comment:
Once you're done here, please remove these asserts. The value being non-null is a structural part of the API now.

Reply:
Should I delete all the null-check asserts that I added after testing and before pushing? I noticed that they should never fire unless I missed an upstream fix.

Reply:
Sure. They're a good refactoring signal, but they're going to become noise once this is all finished.